### ISSUE TITLE
[Request] A few suggestions

### ISSUE THREAD

**Issue by Hello-World-Traveler** [nandyalu/trailarr#59]
There has been a huge need for this in the arr community. thank you.

 I have just notice a movie trailer download for a movie that is still untitled. The video was photos of a bitten hand. 

1. Add approve channel list 
2. Have a minimum size in settings
3. Anything Untitled in the name shouldn't be downloaded.
4. Add pending on movie page if trailer is being downloaded.
5. Add Session Timed Out timer settings 
6. Second page in logs for movies download such as:

```

10/10/2024
Movie 1 (2022)
Movie 2 (2024)
Movie 3 (2008)
04/10/2024
Movie 1 (2021)

```
This way you can see what and when something was downloaded.

**Comment by nandyalu** [comment]
- Add approve channel list

This can be done, but I feel like this will cause more confusion. Do you think it would be better to maintain a list of approved channels on github and make app use that or make this an option to set by user?

- Have a minimum size in settings

Minimum size would not work, yt-dlp does not provide any such option... but, a minimum duration can be set. Trailarr already makes sure videos downloaded are less than 10 minutes (or 5, need to check release notes).

- Anything Untitled in the name shouldn't be downloaded.

Trailarr already filters out videos with "review" in title. We could add other filters, maybe make it an option.

- Add pending on movie page if trailer is being downloaded.

Any media that says "Monitored" is pending a trailer download. 

- Add Session Timed Out timer settings

Current timeout is 10 minutes. This can be added, but is it really needed?

- Second page in logs for movies download such as

I am planning to add something like this. But it's gonna take time.

**Comment by Hello-World-Traveler** [comment]
> This can be done, but I feel like this will cause more confusion. Do you think it would be better to maintain a list of approved channels on github and make app use that or make this an option to set by user?

list of approved channels on github could be the way to go.

> Trailarr already filters out videos with "review" in title. We could add other filters, maybe make it an option.

The script downloaded a trailer for an untitled movie.  Could be an option for the user?

>Current timeout is 10 minutes. This can be added, but is it really needed?

I was editing the settings and stepped away, the page was wiped. If I had the option I would turn it off but that's just me.

It's a good and much needed project!

**Comment by Hello-World-Traveler** [comment]
Does this include Sonarr and Radarr settings? It's not a big deal about the time out, it can become an annoyance.

Feel free to close this when needed.

**Comment by nandyalu** [comment]
> Does this include Sonarr and Radarr settings? It's not a big deal about the time out, it can become an annoyance.

No, adding connection does not save automatically. Those are saved only when you hit submit. So that Trailarr does not create duplicate connections.

**Comment by Hello-World-Traveler** [comment]
A small list of approved YouTube channels
```
Netflix https://www.youtube.com/channel/UCWOA1ZGywLbqmigxE4Qlvuw
Unvisal pictures  https://www.youtube.com/channel/UCq0OueAsdxH6b8nyAspwViw
MAX https://www.youtube.com/channel/UCx-KWLTKlB83hDI6UKECtJQ
Liongates Movies https://www.youtube.com/channel/UCJ6nMHaJPZvsJ-HmUmj1SeA
Disney plus https://www.youtube.com/channel/UCIrgJInjLS2BhlHOMDW7v0g
Apple TV https://www.youtube.com/channel/UC1Myj674wRVXB9I4c6Hm5zA
Sony Pictures https://www.youtube.com/channel/UCz97F7dMxBNOfGYu3rx8aCw
Hulu https://www.youtube.com/channel/UCE5mQnNl8Q4H2qcv4ikaXeA
```

**Comment by Kamikaze01git** [comment]
Sorry to revive this old thread.
But how does this list work? I am mainly looking for German trailers, and it would be very(!) helpful for me if I could restrict Trailarr to specific YouTube channels. Very often, when searching manually, a German trailer with the same search term appears right at the top as the first result, but yt-dlp does not select this trailer despite using the same search string.

The hit rate for foreign-language trailers could be significantly improved if:

1. The profile allowed specifying that downloads are only permitted from certain YouTube channels.
2. In that context, it were also possible to configure/increase the number of search results taken into account (it may be that the German trailer from that particular channel only appears, for example, as the 15th search result).

With these options, it would presumably be possible to find and download trailers in a specific language much more precisely. Point 1 alone would already be worth its weight in gold!

:-)

**Comment by nandyalu** [comment]
I'm definitely open to making Trailarr work better for non-english trailers. 

So, what I understand is that adding list of youtube channels would definitely help - NOTED.

For increasing search limit, it is definitely possible, but increases load and calls to YouTube (which is not a good thing). Can you try doing the search for one of the German trailers, and see what shows up in logs (Debug logs) - why did Trailarr reject them? Was there a German trailer in the results?

One thing we need to keep in mind is that, searching logic is not native to Trailarr logic and is reliant on yt-dlp logic, which is very much restricted.

**Comment by Kamikaze01git** [comment]
Hmm... maybe this is already possible?
There is an option in the profile section for "[Yt-dlp Extra Options](https://nandyalu.github.io/trailarr/user-guide/settings/profiles/settings/search/#:~:text=will%20be%20skipped.-,Yt%2Ddlp%20Extra%20Options,-Type)"
And as we can see in yt-dlp description there is a flag [--match filters](https://github.com/yt-dlp/yt-dlp#usage-and-options:~:text=same%20as%20%2D%2Ddate-,%2D%2Dmatch%2Dfilters,-FILTER%20%20%20%20%20%20%20%20%20%20Generic%20video)

My prefered channel list would be:
```
UCOL10n-as9dXO2qtjjFUQbQ   # KinoCheck / KinotrailerDE
UCqvlLgRFzhlUiOK_IhiQxkQ   # Filmtoast
UCWOw3LaCiK5-UkwoVSrg7eg   # Moviepilot
UCR88edMaCp9veFK6PVXyYpQ   # Warner Bros. Germany
UCPqZp43m3Cg69JkvCvXh8Eg   # Universal Pictures Germany
UCV297SPE0sBWzmhmACKJP-w   # KinoCheck Heimkino
```

So should i be able to use the "yt-dlp extra option" in the profile an simply put something like this inside?!? Will this work?
```
--match-filters "uploader_id = UCOL10n-as9dXO2qtjjFUQbQ" \
--match-filters "uploader_id = UCqvlLgRFzhlUiOK_IhiQxkQ" \
--match-filters "uploader_id = UCWOw3LaCiK5-UkwoVSrg7eg" \
--match-filters "uploader_id = UCR88edMaCp9veFK6PVXyYpQ" \
--match-filters "uploader_id = UCPqZp43m3Cg69JkvCvXh8Eg" \
--match-filters "uploader_id = UCV297SPE0sBWzmhmACKJP-w"
```

Or (even better): yt-dlp [accepts regex](https://github.com/yt-dlp/yt-dlp#usage-and-options:~:text=*%3D%20(contains)%2C-,~%3D%20(matches%20regex,-)%20and%20following%20string) with ~=
So i can simply use one singe line:
```
--match-filters "uploader_id ~= '(UCOL10n-as9dXO2qtjjFUQbQ|UCqvlLgRFzhlUiOK_IhiQxkQ|UCWOw3LaCiK5-UkwoVSrg7eg|UCR88edMaCp9veFK6PVXyYpQ|UCPqZp43m3Cg69JkvCvXh8Eg|UCV297SPE0sBWzmhmACKJP-w)'"
```

or **FINAL VERSION** even better with ^ at the beginning and $ in the end to exactly(!) match the ID (not only include)
```
--match-filters "uploader_id ~= '^(UCOL10n-as9dXO2qtjjFUQbQ|UCqvlLgRFzhlUiOK_IhiQxkQ|UCWOw3LaCiK5-UkwoVSrg7eg|UCR88edMaCp9veFK6PVXyYpQ|UCPqZp43m3Cg69JkvCvXh8Eg|UCV297SPE0sBWzmhmACKJP-w)$'"
```

:-) Any thing against this way?
Maybe i can use this option to accept only these channels....?

**Comment by nandyalu** [comment]
Technically yes, you can use that (I confirmed that the field has a max length of 500, should be enough). 

For making it easier for everyone to use, I can make this a list you can easily add instead of the regex. I need to find time to get this implemented.

@Kamikaze01git I appreciate the help. Thank you 👍

**Comment by nandyalu** [comment]
@Kamikaze01git This issue is old at this point, do we still need the youtube filters. Are you seeing incorrect trailers downloaded by Trailarr?

The plan is to add TMDB integration into Trailarr next and use the trailers from TMDB before searching directly, need to figure out how to integrate without breaking existing options, but it's in the plan. I am thinking this becomes less relevant once TMDB integration is in place. Thoughts?

**Comment by Kamikaze01git** [comment]
Hello.

Thank you very much for continuing to work so diligently on this project. It is without a doubt still the number one solution on the internet when it comes to clean trailer management and integration into an existing ARR system. That is truly great work!

Unfortunately, with family, kids, work, and everything else, I’m very busy, so I tend to let these things slide sometimes.

Back then, I already reported on Discord that I was getting an error when using the yt-dlp extra option `--match-filters`.

```
Cannot override default YT-DLP option 'match_filter' provided in Profile
```

At the time, you already explained that Trailarr itself uses this option internally for `include words`, `exclude words`, and the `duration filter`.
That is understandable and makes perfect sense.

I didn’t pursue it any further afterward, disabled automatic trailer downloads, and only downloaded trailers manually when needed using the correct ID.

*Regarding your actual question:*
TMDB would of course be fantastic for people like me who are looking for German trailers. The requirement, naturally, is that I can specify the language. Right now, TMDB’s web interface allows you to select a language, and it then shows trailers in that language. The language selection even explicitly uses a second language as a fallback.


<img width="1279" height="450" alt="Image" src="https://github.com/user-attachments/assets/278dc5fb-4c95-4a94-a4ac-ab03edf55e78" />



So far, I’ve only used TMDB for fetching plots, cast information, etc. (the ARR system handles that automatically). But having a matching trailer alongside it would definitely be another level of improvement.

**Long story short:
If this can be integrated cleanly into Trailarr, it would be a real game-changer and would make the issue mentioned above obsolete ;-)**

**Comment by nandyalu** [comment]
Just added an option to `Profiles` for `uploader_ids` that accepts a comma-separated list of values of both the uploader handles (e.g. `@WarnerBrosPictures`) and/or channel IDs (from the channel url).

---
> Thank you very much for continuing to work so diligently on this project. It is without a doubt still the number one solution on the internet when it comes to clean trailer management and integration into an existing ARR system. That is truly great work!

Thank You! That means a lot.

---

Regarding the TMDB integration, I don't yet know what all TMDB API provides but I want to make sure you can select the language and trailer/teaser/clip/behind the scenes/bloopers/featurette that are available.

<img width="203" height="247" alt="Image" src="https://github.com/user-attachments/assets/a734b046-c0fe-4519-8ac2-699e01848837" />

**Comment by Kamikaze01git** [comment]
Thank you so much for your work. You’re just casually pushing out updates, cleaning up the UI, and adding the ability to restrict profiles to specific YouTube channels or IDs.

I just tested it with 3 trailers (I don’t have more time at the moment), and all 3 were downloaded correctly in German immediately on the very first try :)))

For me and my use case, this feels like an absolute game changer right now.

I’m going to take a closer look at it, but if this works well and reliably, I’ll probably let your masterpiece (and yes, that’s exactly what it is!!) run across my entire collection together with a YouTube cookie ;-)

So from my side: deep gratitude. Trailarr is taking my collection to the next level!

**Comment by Hello-World-Traveler** [comment]
@nandyalu finally got back to this. Few more suggestions;
On movie page:
1) movie ratings 
2) above watch button, name of youtube channel the link goes too.

Auto Rescan files in movie folder to show up in Downloads area after download/convert.
Recent download page that list the last 20 downloads to view, play,edit
Test GPU Acceleration on settings, this will insure the users GPU can convert video. I think there is fallback to non GPU Acceleration however GPU would always be preferred.
